### DISCUSSION TITLE
ESP Encapsulation in TCP/TLS - RFC 8229

### DISCUSSION THREAD

**Discussion by davewj100** [strongswan/strongswan#357] [Q&A]
Good day,

I have a client where I'd like to implement IPSec in Strongswan over TCP (preferably TLS) as described in RFC 8229. This is to pass content inspection gateways which only allow 443/TLS outbound.

I found this completed feature request #2189 that describes it: https://wiki.strongswan.org/issues/2189

But I can't find how to make use of it. What do I change in ipsec.conf or charon.conf (or somewhere else) to enable ESP encapsulation in TLS?

Thanks so much!

Dave J

**Comment by Thermi** [comment]
You can't because support for it is not integrated into strongSwan yet.

**Reply by TomHsiung** [reply]
How about this feature today?
